### PR TITLE
Refactor ChromeManifestScanner: extend BaseScanner

### LINKED ISSUE #144: ChromeManifestScanner should extend BaseScanner
As mentioned in #133, we can have the `ChromeManifestScanner` extend `BaseScanner` as well with minimal effort, but we'll do it in a separate issue.

### PR DESCRIPTION
**Pull request by tofumatt**
Closes #144.

### CODE REVIEW

**Comment by tofumatt** [comment]
I cleaned up a few classes while I was at it; hope you don't mind it being mixed in with this commit.

File: tests/scanners/test.chromemanifest.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import ChromeManifestScanner from 'scanners/chromemanifest';
 import * as rules from 'rules/chromemanifest';
 import { getRuleFiles, validChromeManifest } from '../helpers';
-import { ignorePrivateFunctions } from 'utils';
+import { ignorePrivateFunctions, unexpectedSuccess } from 'utils';
```
Comment (muffinresearch) [inline tests/scanners/test.chromemanifest.js]: This shouldn't work as it *is* in helpers. Re: #187

Comment (tofumatt) [inline tests/scanners/test.chromemanifest.js]: Oh weird, I saw it imported this way elsewhere.

Comment (muffinresearch) [inline tests/scanners/test.chromemanifest.js]: oh, ok - that's a definite bug/mistake then.

File: src/scanners/chromemanifest.js [hunk 2]
```diff
@@ -1,32 +1,21 @@
...
 
-  scan(_ChromeManifestParser=ChromeManifestParser, _rules=rules) {
-    return new Promise((resolve) => {
-      var validatorMessages = [];
-      var cmParser = new _ChromeManifestParser(this.stream, this.filename);
-      return cmParser.parse()
+  _getContents(_ChromeManifestParser=ChromeManifestParser) {
+    return new Promise((resolve, reject) => {
+      var cmParser = new _ChromeManifestParser(this.contents, this.filename);
+      cmParser.parse()
```
Comment (muffinresearch) [inline src/scanners/chromemanifest.js]: Can't you just return this and not create a new Promise?

**Comment by muffinresearch** [comment]
I think you can kill the unnecessary Promise. Otherwise r+wc

File: src/scanners/chromemanifest.js (hunk 2 above)
Comment (tofumatt) [inline src/scanners/chromemanifest.js]: Could do; we won't catch a parse error but I suppose it doesn't even implement `reject` right now anyway.

**Comment by tofumatt** [comment]
Sweet. screenshot 2015-11-03 16 21 54

File: src/scanners/chromemanifest.js (hunk 2 above)
Comment (muffinresearch) [inline src/scanners/chromemanifest.js]: But if `parse()` blows up that'll implicitly reject the returned promise won't it?

**Comment by muffinresearch** [comment]
more-red-than-green-award

**Pull request merged**